### PR TITLE
[luci-interpreter] Support SelectV2 op

### PR DESCRIPTION
**Pull request by ragmani**
This makes luci-interpreter support SelectV2 op.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

---

For https://github.com/Samsung/ONE/issues/12148

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/Select.h [hunk 1]
```diff
@@ -42,8 +42,6 @@ class Select : public Kernel
   void evalFloat() const;
 
 private:
-  // for SelectV2
-  bool _requires_broadcast = false;
```
[the remark this answers is not in the document]
Comment (ragmani) [reply compiler/luci-interpreter/src/kernels/Select.h:46]: OK I will separate it.

**Review by seanshpark — DISMISSED** [review]
LGTM thank you!

**Comment by ragmani** [comment]
@jinevening PTAL

File: compiler/luci-interpreter/src/kernels/SelectV2.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      evalFloat();
+      break;
+    default:
+      throw std::runtime_error("luci-intp SelectV2 unsupported type.");
+  }
+}
+
+void SelectV2::evalFloat() const
+{
+
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/SelectV2.cpp:73]:
```suggestion
```

File: compiler/luci-interpreter/src/kernels/SelectV2.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  auto cond_shape = condition()->shape();
+  auto cond_num_dims = cond_shape.num_dims();
+  auto t_shape = t()->shape();
+  auto e_shape = e()->shape();
+
+  bool is_input_condition_scalar = cond_num_dims == 0;
+  bool has_rank_one_input_condition = cond_num_dims == 1 && cond_shape.dim(0) == t_shape.dim(0);
+
+  _has_low_rank_input_condition =
+    (is_input_condition_scalar || has_rank_one_input_condition) && t_shape == e_shape;
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/SelectV2.cpp:53]: This behavior is different from tflite kernel.

```suggestion
  _has_low_rank_input_condition =
    is_input_condition_scalar || has_rank_one_input_condition;
```

`t_shape == e_shape` is necessary? It seems that `has_low_rank_input_condition` is only used by select v1.

Comment (ragmani) [reply compiler/luci-interpreter/src/kernels/SelectV2.cpp:53]: > `t_shape == e_shape` is necessary?

Yes, it is necessary if select v2 kernel uses `RankOneSelect()`.

> `has_low_rank_input_condition` is only used by select v1.

I just implemented it following below conditions. 

### TFL SelectV2 op

https://www.tensorflow.org/mlir/tfl_ops#tflselect_v2_tflselectv2op

Select values of 'x' if the corresponding value of 'condition' is true or the value of 'y' if false. There are valid condition input sizes:

1. Either the same shape (in which case the select is elementwise), or
2. Broadcastable shapes between 'condition', 'x' and 'y'.

Anyway, I will change this kernel to use only `BroadcastSelect4DSlow()` as your advice.

File: compiler/luci-interpreter/src/kernels/SelectV2.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+void SelectV2::configure()
+{
+  LUCI_INTERPRETER_CHECK(condition()->element_type() == DataType::BOOL);
+  LUCI_INTERPRETER_CHECK(t()->element_type() == e()->element_type());
+  LUCI_INTERPRETER_CHECK(t()->element_type() == output()->element_type());
+
+  auto cond_shape = condition()->shape();
+  auto cond_num_dims = cond_shape.num_dims();
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/SelectV2.cpp:44]: This variable is unused.

```suggestion
```

**Review by jinevening — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**